### PR TITLE
Completing Supporting Documents

### PR DESCRIPTION
**Pull request by z3dev**
## WORK IN PROGRESS

- [X] Update README
- [X] Add CONTRIBUTING
- [X] Document changes to API
  - Imported library from openjscad
  - Wrapped for NPM usage
  - Corrected Vector3D() in order to instantiate from anonymous objects
  - Corrected several default states for CAG.isCanonicalized
  - Corrected missing variable declarations in CAG.rectangle()
  - Added options (symmetrical) to extrudeInOrthonormalBasis and extrudeI……nPlane
  - Added CSG.cylinderElliptic()
  - Added automatic creation of API documentation via JSDOC
- [ ] Decide where to put the API documentation

Once we have this done, then I'll apply similar changes to the other libraries.

### CODE REVIEW

**Comment by kaosat-dev** [comment]
A few thoughts:

API: 
- great idea about 'Document changes to API': I think we should put that in a changelog file :)
- side note, there are ways to autogenerate that from our git commits, but we would have to change how we merge PRS a bit
- the generated docs should be delivered with the repo, not generated manually by the user (we want to make things easy for end users) , but perhaps in a clearer manner than an 'out' folder ?

About CONTRIBUTING
- it is already "[standardized](https://github.com/blog/1184-contributing-guidelines)" on Github so we should stick to what works ie a contributing.md file
- best to avoid wiki pages, stick to md files (less of 'out of repostiory' data)

About README
- list of maintainers : yeah I don't think we need that for now ?

**Comment by z3dev** [comment]
CONTRIBUTING
- Available now. Please review.
- Maybe the first section should move to the README, leaving development in CONTRIBUTING
- There are still links to WIKI pages for Reporting Issues guide, and Contributing guide.

README
- Looks pretty good.
- Shall the API generation be dropped? i think so

API DOCS
- Would be cool if GitHub could support this but...
- Maybe www.openjscad.org can house this as another set of pages?
- Rename 'out' to 'doc'?

**Comment by z3dev** [comment]
Looking good.

Issues / Bugs
- Suggest adding templates then an offline document isn't required

API Documentation
- Documentation should be automatically generated for each release increment
- Where to put the documentation?. Any ideas?
  - Put the HTML/CSS on GitHub? I don't think this works
  - Put the HTML/CSS on www.openjscad.org ? Might work but we need access

**Comment by kaosat-dev** [comment]
Very nice progress !
- out to doc is good
- agreed on the issue template, let's do that!
- for the docs, perhaps a good solution is using the gh-pages branch to publish it as a mini website

File: jsdoc.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "opts": { // command line options
```
Comment (kaosat-dev) [inline jsdoc.json]: Javascript comments are not valid in a json document

File: jsdoc.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
     "template": "templates/default",
     "encoding": "utf8",
-    "destination": "./out/",
+    "destination": "./doc/",
     "recurse": true,
     "readme": "README.md"
     //"tutorials": "path/to/tutorials", // same as -u path/to/tutorials
```
Comment (kaosat-dev) [inline jsdoc.json]: same here

File: package.json [hunk 3]
```diff
@@ -53,7 +53,8 @@
   "devDependencies": {
     "ava": "^0.19.1",
     "babel-preset-es2015": "^6.24.1",
-    "babel-register": "^6.24.1"
+    "babel-register": "^6.24.1",
+    "jsdoc": "^3.4.3"
```
Comment (kaosat-dev) [inline package.json]: groovy :)

File: README.md [hunk 4]
```diff
@@ -36,13 +35,25 @@ const csg = require('csg')
...
-TODO
+And open doc/index.html using a browser.
 
-## Contribute
+Also see the [OpenJsCad User Guide](https://en.wikibooks.org/wiki/OpenJSCAD_User_Guide).
 
 For questions about the API, please contact the [User Group](https://plus.google.com/communities/114958480887231067224)
 
-PRs accepted.
+## Contribute
```
Comment (kaosat-dev) [inline README.md]: still cannot make up my mind about where this belongs

**Comment by z3dev** [comment]
API Docs

Great! Let's try to use gh-pages. It's a good solution even if we have to publish manually.

**Pull request merged**